src/android/GroupAccessor.java

```diff
@@ -305,4 +307,71 @@ protected String removeLabelFromData(String sourceId) {
         return finalGroups;
     }
 
+    /**
+     * Adds new label with given title on every account on device and syncs them all
+     *(for generating some autogenerating info).
+     *
+     * @param title Title of label wanted to be added.
+     * @return success message if succeed and exception message if failed
+     */
+    protected String addLabelToDatabase(String title) {
+        ArrayList<ContentProviderOperation> ops =
+                new ArrayList<>();
+        AccountManager accountManager = AccountManager.get(app.getActivity());
+        Account[] accounts = accountManager.getAccounts();
+        for (int i = 0; i < accounts.length; i++) {
+            ops.add(ContentProviderOperation.newInsert(ContactsContract.Groups.CONTENT_URI)
+                    .withValue(ContactsContract.Groups.TITLE, title)
+                    .withValue(ContactsContract.Groups.ACCOUNT_NAME, accounts[i].name)
+                    .withValue(ContactsContract.Groups.ACCOUNT_TYPE, accounts[i].type)
+                    .withYieldAllowed(true)
+                    .build()
+            );
+        }
+
+        try {
+            ContentProviderResult[] results = app.getActivity().getContentResolver().applyBatch(ContactsContract.AUTHORITY, ops);
+            if (results.length >= 1) {
+                //Syncs accounts for generate SOURCE_ID, SYNC2, SYNC3
+                GroupHelper.requestSyncNow(app.getActivity());
+            }
+        } catch (RemoteException e) {
+            e.printStackTrace();
+            e.getMessage();
+        } catch (OperationApplicationException e) {
+            e.printStackTrace();
+            return e.getMessage();
+        }
+        return QUsersCordova.SUCCESS;
+    }
+
+    /**
+     * Changes existing labels title into given one.
+     *
+     * @param sourceId Existing label's sourceId which title wanted to be changed
+     * @param title New title
+     * @return success message if succeed and exception message if failed
+     */
+    protected String editLabelInDatabase(String sourceId, String title) {
+        ArrayList<ContentProviderOperation> ops =
+                new ArrayList<>();
+        ops.add(ContentProviderOperation.newUpdate(ContactsContract.Groups.CONTENT_URI)
+                .withSelection(ContactsContract.Groups.SOURCE_ID + "=?", new String[]{sourceId})
+                .withValue(ContactsContract.Groups.TITLE, title)
+                .withYieldAllowed(true)
+                .build()
+        );
+
+        try {
+            ContentProviderResult[] results = app.getActivity().getContentResolver().applyBatch(ContactsContract.AUTHORITY, ops);
+        } catch (RemoteException e) {
+            e.printStackTrace();
+            e.getMessage();
```

Comment:
Why you skip this Exception? Follow your logic you should return message of Error. If the function throws RemoteException,  it will return QUsersCordova.SUCCESS